From 77be4f28c1c20eea6aad9854a668fadadcd159ba Mon Sep 17 00:00:00 2001
From: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Date: Thu, 22 Aug 2019 19:29:31 -0700
Subject: [PATCH 0659/1851] v4l: xilinx: vip: Keep NV16 first in vip table

commit 9335cb3d561497603c2e4ec2e5f36f1e428b314b from
https://github.com/Xilinx/linux-xlnx.git

When Scene change detection IP alone is in the v4l2 pipeline,
the driver is unable to enumerate all the supported pixel formats
based on media code set on the subdev source port. The function
xvip_dma_enum_format() just returns the pix format whichever comes
first in the vip table list.

This patch ensures that NV16 comes first in the vip table. This fixes
the issue temporarily as SCD IP supports only NV16 in 422 8bit
formats.

The permanant fix is to implement a generic interface between V4L2
bridge driver and dma driver to get the supported pixel formats.

Signed-off-by: Satish Kumar Nagireddy <satish.nagireddy.nagireddy@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-vip.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/media/platform/xilinx/xilinx-vip.c b/drivers/media/platform/xilinx/xilinx-vip.c
index d0e8357fda27..5de2f90df593 100644
--- a/drivers/media/platform/xilinx/xilinx-vip.c
+++ b/drivers/media/platform/xilinx/xilinx-vip.c
@@ -40,14 +40,14 @@ static const struct xvip_video_format xvip_video_formats[] = {
 	  2, 12, V4L2_PIX_FMT_X016, 2, 1, 2, 2 },
 	{ XVIP_VF_YUV_420, 16, NULL, MEDIA_BUS_FMT_UYYVYY16_4X32,
 	  2, 12, V4L2_PIX_FMT_X016M, 2, 2, 1, 2 },
-	{ XVIP_VF_YUV_422, 8, NULL, MEDIA_BUS_FMT_UYVY8_1X16,
-	  2, 16, V4L2_PIX_FMT_YUYV, 1, 1, 2, 1 },
-	{ XVIP_VF_VUY_422, 8, NULL, MEDIA_BUS_FMT_UYVY8_1X16,
-	  2, 16, V4L2_PIX_FMT_UYVY, 1, 1, 2, 1 },
 	{ XVIP_VF_YUV_422, 8, NULL, MEDIA_BUS_FMT_UYVY8_1X16,
 	  1, 16, V4L2_PIX_FMT_NV16, 2, 1, 1, 1 },
 	{ XVIP_VF_YUV_422, 8, NULL, MEDIA_BUS_FMT_UYVY8_1X16,
 	  1, 16, V4L2_PIX_FMT_NV16M, 2, 2, 1, 1 },
+	{ XVIP_VF_YUV_422, 8, NULL, MEDIA_BUS_FMT_UYVY8_1X16,
+	  2, 16, V4L2_PIX_FMT_YUYV, 1, 1, 2, 1 },
+	{ XVIP_VF_VUY_422, 8, NULL, MEDIA_BUS_FMT_UYVY8_1X16,
+	  2, 16, V4L2_PIX_FMT_UYVY, 1, 1, 2, 1 },
 	{ XVIP_VF_YUV_422, 10, NULL, MEDIA_BUS_FMT_UYVY10_1X20,
 	  1, 16, V4L2_PIX_FMT_XV20, 2, 1, 2, 1 },
 	{ XVIP_VF_YUV_422, 10, NULL, MEDIA_BUS_FMT_UYVY10_1X20,
-- 
2.31.1

